jwst/ami/ami_analyze_step.py

```diff
@@ -24,7 +24,7 @@ class AmiAnalyzeStep(Step):
         usebp = boolean(default=True) # If True, exclude pixels marked DO_NOT_USE from fringe fitting
         firstfew = integer(default=None) # If not None, process only the first few integrations
         chooseholes = string(default=None) # If not None, fit only certain fringes e.g. ['B4','B5','B6','C2']
-        affine2d = string(default=None) # ASDF file containing user-defined affine parameters
+        affine2d = string(default='commissioning') # ASDF file containing user-defined affine parameters OR 'commssioning'
```

Comment:
Quick question - did you intend to still support None as an option here?  Now that it is no longer default, I don't think there's a way to set it to None on the command line.  This:
`strun jw04478001001_03102_00001_nis_cal.fits --affine2d=None`
throws an error because it expects an ASDF file called "None".  Maybe add some special handling for None as a string, too?

Reply:
Sorry, I was calling this wrong and got confused -- it does default to None when the input is "None", so no special handling needed there.

Reply:
Good catch. Hmm, my thinking was that we just want it to default to None internally if there are issues with the ASDF file, where it will then use an identity transform + a rotation search (the `if affine2d is None` block in `apply_LG_plus()`). So to force it to use the rotation search option they would have to provide a garbage string to trigger the exceptions in `override_affine2d()`... I'll think about this a bit more, and I've just sent a message to my AMI colleagues in case they have a different opinion.

Reply:
Ok, that's great! For some reason I didn't see your second comment before replying. I think in general we expect users will be using the "commissioning" default or supplying their own ASDF file but it's good that the previous behavior (affine2d=None which leads to a rotation-only transform) is still accessible.

Reply:
Checking again, and making sure I have my environments correct, it does default to None and attempts to continue, but then throws this error when it tries to find the rotation:

```
  File "jwst/ami/ami_analyze.py", line 162, in apply_LG_plus
    affine2d = find_rotation(
               ^^^^^^^^^^^^^^
  File "jwst/ami/find_affine2d_parameters.py", line 101, in find_rotation
    jw = lg_model.LgModel(mask='jwst_ami', holeshape=holeshape, over=over, affine2d=aff)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "jwst/ami/lg_model.py", line 101, in __init__
    mask = mask_definition_ami.NRM_definition(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: NRM_definition.__init__() missing 1 required positional argument: 'nrm_model'

```

Reply:
Oh I see the issue. I might remove the string `mask` input after all then and just pass the NrmModel containing the mask info through directly to LgModel.

Reply:
I made the changes and tried it locally, it seems to do the right thing when I call it e.g.
```
calints = 'jw01093015001_03102_00001_nis_calints.fits'
analyze = AmiAnalyzeStep()
analyze.affine2d = None
outmodels = analyze.run(calints)
```
though I got a whole mess of errors ending in a `UnicodeDecodeError` when I tried the `strun` version. How do you call it correctly with None as an argument?

Reply:
I'm not sure you can pass a null value to a string argument via the command line.  I think you would need to add handling for an input value of the string "None" - just translate it internally to the null value `None`.

Reply:
I tried to add some lines (just pushed) handling a possible "None" string input but am getting the same error -- maybe I'm still not calling it right? I tried:
`strun /ifs/jwst/wit/niriss/rcooper/nis_019/pipeline_calibrated_data_02062024/jw01093015001_03102_00001_nis_calints.fits --affine2d None`

and with `--affine2d=None` and `--affine2d "None"` and pretty much all variations I could think of but still get `ValueError: Does not appear to be a ASDF file`. It seems that it doesn't get to the lines I added before failing because the error originates in stpipe/config_parser.py. I thought step arguments could be passed like that to strun but it seems like it only wants a config file? Is that the ASDF file it's looking for here?

Reply:
It looks like you're calling it the same way I accidentally did! You need to run it as `strun ami_analyze filename.fits` -- testing this locally, it looks to me like your new changes are working now to accept "None"

Reply:
...but now I see an error with the default "commissioning" case:
```
  File "jwst/ami/ami_analyze_step.py", line 211, in process
    oifitsmodel, oifitsmodel_multi, amilgmodel = ami_analyze.apply_LG_plus(
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "jwst/ami/ami_analyze.py", line 186, in apply_LG_plus
    log.info(f'\trotradccw={affine2d.rotradccw:.4f}')
                           ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unsupported format string passed to NoneType.__format__

```

Reply:
Love it when an extra print statement breaks everything 🫠 Just fixed that.

Reply:
I agree, I'll look into it and see if there's something I'm missing, or file a ticket if it's something we don't yet support.